src/org/thoughtcrime/securesms/WebxdcActivity.java

```diff
@@ -48,40 +48,69 @@
 import org.thoughtcrime.securesms.util.Util;
 
 import java.io.ByteArrayInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
 import java.io.InputStream;
 
 import java.util.HashMap;
 import java.util.Map;
 
 public class WebxdcActivity extends WebViewActivity implements DcEventCenter.DcEventDelegate  {
   private static final String TAG = WebxdcActivity.class.getSimpleName();
+  private static final String EXTRA_ACCOUNT_ID = "accountId";
+  private static final String EXTRA_APP_MSG_ID = "appMessageId";
+  private static final String EXTRA_HIDE_ACTION_BAR = "hideActionBar";
   private static final int REQUEST_CODE_FILE_PICKER = 51426;
+
   private ValueCallback<Uri[]> filePathCallback;
   private DcContext dcContext;
   private DcMsg dcAppMsg;
   private String baseURL;
   private String sourceCodeUrl = "";
   private boolean internetAccess = false;
+  private boolean hideActionBar = false;
+
+  public static void openMaps(Context context, int chatId) {
+    DcContext dcContext = DcHelper.getContext(context);
+    int msgId = dcContext.initWebxdcIntegration(chatId);
+    if (msgId == 0) {
+      try {
+        InputStream inputStream = context.getResources().getAssets().open("webxdc/maps.xdc");
+        String outputFile = DcHelper.getBlobdirFile(dcContext, "maps", ".xdc");
+        Util.copy(inputStream, new FileOutputStream(outputFile));
+        dcContext.setWebxdcIntegration(outputFile);
+        msgId = dcContext.initWebxdcIntegration(chatId);
+      } catch (IOException e) {
+        throw new RuntimeException(e);
+      }
```

Comment:
that would let the app crash, even though it should not happen in normal usable at all, maybe better be a bit more friendly and give some hints:

```suggestion
      } catch (IOException e) {
        e.printStackTrace();
      }
      if (maps == 0) {
         Toast.makeText(context, "Cannot get maps.xdc, see log for details.", Toast.LENGTH_LONG).show();
         return;
      }
```